server/routes/visitJourney/dateAndTime.test.ts

```diff
@@ -531,59 +508,59 @@ describe('Update journey specific warning messages', () => {
       .expect('Content-Type', /html/)
       .expect(res => {
         const $ = cheerio.load(res.text)
+        expect($('[data-test="slot-change-reason"]').length).toBe(0)
         expect($('[data-test="restriction-change-reason"]').length).toBe(0)
         expect($('input#1').prop('checked')).toBe(true)
       })
   })
 
-  it('should show message with no slot selected when visit has changed from open to closed (visitor restriction), but original timeslot unavailable', () => {
+  it('should select original slot with no messages if no restriction change and original time available (even if overbooked)', () => {
     currentlyBookedSlot.visitRestriction = 'OPEN'
 
-    currentlyAvailableSlots[0].availableTables = -1 // test over-booked (which includes fully-booked)
-    currentlyAvailableSlots[0].visitRestriction = 'CLOSED'
+    // Allowing over-booing is OK because the original visit (being updated) is one of the already-booked spaces
+    currentlyAvailableSlots[0].availableTables = -1
+    currentlyAvailableSlots[0].visitRestriction = 'OPEN'
 
-    visitSessionData.visitRestriction = 'CLOSED'
-    visitSessionData.closedVisitReason = 'visitor'
+    visitSessionData.visitRestriction = 'OPEN'
 
     return request(sessionApp)
       .get('/visit/ab-cd-ef-gh/update/select-date-and-time')
       .expect(200)
       .expect('Content-Type', /html/)
       .expect(res => {
         const $ = cheerio.load(res.text)
-        expect($('[data-test="restriction-change-reason"]').text()).toContain(
-          'A new visit time must be selected as this is now a closed visit due to a visitor restriction.',
-        )
-        expect($('input:checked').length).toBe(0)
+        expect($('[data-test="slot-change-reason"]').length).toBe(0)
+        expect($('[data-test="restriction-change-reason"]').length).toBe(0)
+        expect($('input#1').prop('checked')).toBe(true)
       })
   })
 
-  it('should show message with no slot selected when visit has changed from open to closed (prisoner restriction), but original timeslot unavailable', () => {
+  it('should show two messages with no slot selected when visit has changed from open to closed and original time slot unavailable', () => {
     currentlyBookedSlot.visitRestriction = 'OPEN'
 
-    currentlyAvailableSlots[0].availableTables = 0
+    currentlyAvailableSlots[0].startTimestamp = '2022-10-17T09:01:00'
```

Comment:
@hutcheonb-moj For info, I've tweaked the tests a bit as I think they weren't quite setting up the scenarios correctly. For each one in this set, I'm checking the presence/absence of **both** of the possible messages. I'm also setting up the situation where the original time slot isn't available by changing the time. I'm not using the `availableTables` as this is effectively ignored - as even if the slot is overbooked we let an update proceed because it's _already_ in the count of bookings. 

Initially I thought the tests were right but there was something wrong with the implementation - I then realised that the implementation worked but the tests weren't quite doing what the ought to :swear